challenges.yml

```
@@ -65,3 +70,118 @@ stimulus_controllers_convention:
+    inside the `static` property `targets`. Then, you can use any of the registered targets by their name post-fixing
+    `Target` to work with the first target found in the template, or `Targets` to get an array of matching targets.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          Register the target like this: `static targets = ['items'];`
+        b: |
+          Change the `this.itemTarget` call to this `this.itemTargets`
+        c: |
+          Change the `this.itemTarget` call to this `this.itemTargetsAll`
+        d: |
+          Change the `this.itemTarget` call to this `this.itemAll`
+      correct: b
+stimulus_actions:
+  question: |
+    What happens if I don't specify the event to listen when defining a Stimulus action?
+    `<button data-action="counter#increment" />`
+  explanation: |
+    If you don't explicitly define which event you want to listen, then, Stimulus will set a default event
+    based on the element's type. For example a `<button>` element would be attached to the `click` event.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          Stimulus will throw an error.
+        b: |
+          Nothing will happen because there aren't any event to listen.
+        c: |
+          Based on the element's type a default event will be attached instead.
+        d: |
+          It's a quick way to listen to all events.
+      correct: c
+event_current_target:
+  question: |
+    What element will be attached to the `event.currentTarget` variable when clicking on the `<i>` tag.
+    ```html
+    <button data-action="user-controller#addUser">
+      <i class="fa fa-plus"></i>
+      <span>Add a new User</span>
+    </button>
+    ```
+  explanation: |
+    The `event.currentTarget` will always hold a reference to the element that we attached the event to. And at
+    the other hand, `event.target` references to the element which received or listened to the event, in this case, the click event.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          The `<button>` element
+        b: |
+          The `<i>` element
+        c: |
+          The `<span>` element
+      correct: a
+js_data_attributes:
+  question: |
+    How can I access to the `color-id` data attribute of an element using plain Javascript?
+  explanation: |
+    Javascript stores all of the data attributes inside the `dataset` property of an element but standardizing its key names.
+    For example the `data-api-url` attribute would be renamed to `someElement.dataset.apiUrl`
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          `$(element).data('colorId');`
+        b: |
+          `$(element).data('color-id');`
+        c: |
+          `element.dataset.color-id`
+        d: |
+          `element.dataset.colorId`
+      correct: d
+stimulus_values:
+  question: |
+    If I have this Stimulus value on a controller. How can I get access to it?
```

```suggestion
    Suppose we have this Stimulus value on a controller: 
```

line break between the question and code block

Then:

```
How can we access the `colorSelected` value?
```